activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -131,15 +131,11 @@ def execute(sql, name = nil)
       def exec_query(sql, name = 'SQL', binds = [], prepare: false)
         result = execute(sql, name)
         @connection.next_result while @connection.more_results?
-        ActiveRecord::Result.new(result.fields, result.to_a)
+        ActiveRecord::Result.new(result.fields, result.to_a) if result
```

Comment:
@kamipo are there tests that cover this case? Are there no callers that rely on the fact that `exec_query` did not return `nil` previously.